src/newgrf_roadtype.cpp

```diff
@@ -45,7 +45,7 @@
 			const Town *t = nullptr;
 			if (IsRoadDepotTile(this->tile)) {
 				t = Depot::GetByTile(this->tile)->town;
-			} else if (IsTileType(this->tile, MP_ROAD)) {
+			} else if (IsTileType(this->tile, MP_ROAD) || IsTileType(this->tile, MP_STATION)) {
```

Comment:
It think `MP_STATION` is too generic, you may want to use `IsRoadStopTile()` (maybe `IsStandardRoadStopTile()` or `IsDriveThroughStopTile()` depending on precision needed)

Reply:
Thanks - IsRoadStopTile() looks like the right one, the original bug affects both standard and drive-through road stops.